sqlmesh/core/state_sync/db/facade.py

```diff
@@ -199,7 +199,10 @@ def promote(
                 )
                 != table_infos[name].qualified_view_name.for_environment(environment.naming_info)
             }
-            if not existing_environment.expired:
+            if (
+                not existing_environment.expired
+                and existing_environment.gateway_managed == environment.gateway_managed
```

Comment:
I don't think this is right place for this. We need this check here instead: https://github.com/TobikoData/sqlmesh/blob/c59288bb53e10211e927224e35c01f99b9cfca2f/sqlmesh/core/state_sync/db/facade.py#L228-L232. 

The goal is to *recreate* all the views. The current block has nothing to do with that